### PR TITLE
Update DESIGN.md to include an Overview and entry point for SAM

### PR DESCRIPTION
**Pull request by jfuss**
*Issue #, if available:*

*Description of changes:*
This will add an Overview section to the DESIGN.md file that explains how SAM works and the entry point into the code base. This is useful to help new people in being able to trace through the code base.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: DESIGN.md [hunk 1]
```diff
@@ -1,3 +1,11 @@
+# Overview
+
+SAM is invoke/called by the CloudFormation Service. CloudFormation recognises the `Transform: AWS::Serverless-2016-10-31` header and invokes the SAM translator. This will then take your SAM template and expand it
```
Comment (mwarkentin) [inline DESIGN.md]: invoke -> invoked? Or maybe just drop and do “SAM is called by...”

**Comment by Nr18** [comment]
Oh this was what i was looking for but the reason why was because i wanted to know how i can help and add some things like a property LogRetention on the `AWS::Serverless::Function` resource that will created a log group.

A generic description how you can add things like that would help me and others to contribute to SAM.

File: DESIGN.md (hunk 1 above)
Comment (jfuss) [reply DESIGN.md]: Updated in next commit.

**Comment by jfuss** [comment]
@PaulMaddox Will be updated in the next commit.

@Nr18 Been discussing on #sam-dev slack channel on adding the support but this also got me thinking of how to add more docs. Will comment in slack my idea around more docs and surfacing this data.

**Comment by mwarkentin** [comment]
👍

**Pull request merged**